plugins/tokenizers/mecab.c

```diff
@@ -1091,12 +1099,21 @@ mecab_next_wakati_format(grn_ctx *ctx,
 
     if (space_len > 0) {
       const char *q = r + space_len;
-      while (q < e && (space_len = grn_isspace(q, encoding))) {
+      if (mecab_wakati_delimiter(r)) {
+        tokenizer->next = q;
+        is_leading_space = true;
+        break;
+      }
+      while (q < e && !mecab_wakati_delimiter(q) &&
+             (space_len = grn_isspace(q, encoding))) {
         q += space_len;
       }
-      tokenizer->next = q;
-      break;
+      if (is_leading_space) {
+        tokenizer->next = q;
+        break;
+      }
     }
+    is_leading_space = false;
```

Comment:
Space should not have to be special handled.
Wouldn't the following code work?

```diff
 
-    if (space_len > 0) {
-      const char *q = r + space_len;
-      if (mecab_wakati_delimiter(r)) {
-        tokenizer->next = q;
-        is_leading_space = true;
-        break;
-      }
-      while (q < e && !mecab_wakati_delimiter(q) &&
-             (space_len = grn_isspace(q, encoding))) {
-        q += space_len;
-      }
-      if (is_leading_space) {
-        tokenizer->next = q;
-        break;
-      }
+    if (mecab_wakati_delimiter(r)) {
+      tokenizer->next = r + 1;
+      break;
     }
-    is_leading_space = false;
   }
```

Reply:
This 57625b2...4ea14e3 code could remove leading/trailing spaces in user-defined token like the ` search engine` or `search engine `.